app/assets/javascripts/initializers/initNotifications.js

```diff
@@ -60,70 +53,77 @@ function fetchNotificationsCount() {
   }
 }
 
-function initReactions() {
-  setTimeout(function() {
-    if (document.getElementById('notifications-container')) {
-      var butts = document.getElementsByClassName('reaction-button');
-      for (var i = 0; i < butts.length; i++) {
-        var butt = butts[i];
-        butt.onclick = function(event) {
-          event.preventDefault();
-          sendHapticMessage('medium');
-          var thisButt = this;
-          thisButt.classList.add('reacted');
-
-          function successCb(response) {
-            if (response.result === 'create') {
-              thisButt.classList.add('reacted');
-            } else {
-              thisButt.classList.remove('reacted');
-            }
-          }
+// map over the array of elements and apply a callback function to click event
+function addOnClickHandlerToButtons(className, callBack) {
+  var butts = document.getElementsByClassName(className);
+  var i;
+  var butt;
+  for (i = 0; i < butts.length; i += 1) {
+    butt = butts[i];
+    butt.addEventListener('click', callBack);
+  }
+}
 
-          var formData = new FormData();
-          formData.append('reactable_type', thisButt.dataset.reactableType);
-          formData.append('category', thisButt.dataset.category);
-          formData.append('reactable_id', thisButt.dataset.reactableId);
+function onClickReactionButton(event) {
+  event.preventDefault();
+  sendHapticMessage('medium');
+  var thisButt = this;
```

Comment:
@netochaves this was the cause of the reaction buttons not working.

Reply:
Oh, I see now. I've refactored some functions to arrow functions and because of that, I've changed `this` to `event.target` on a few places.